### ISSUE TITLE
Parsing error: Unexpected token, expected ":"

### ISSUE THREAD

**Issue by Tom910** [typescript-eslint/typescript-eslint#962]
I'm doing migrate of tslint on eslint and get many parsing errors. I used a mixed config for js and ts files. If it is a problem not in my config I create a new issue with another parsing errors

**What code were you trying to parse?**

```ts
export type SyncJsonFile = ({
  path,
  newContent,
}: {
  path: string;
  newContent: any;
}) => Promise<void>;
```

**What did you expect to happen?**
Do not show errors

**What actually happened?**
```
  6:7  error  Parsing error: Unexpected token, expected ":"

  4 |
  5 | export type SyncJsonFile = ({
> 6 |   path,
    |       ^
  7 |   newContent,
  8 | }: {
  9 |   path: string;
```

**Config**
```ts
export const typescriptConfig = {
  extends: ['plugin:@typescript-eslint/eslint-recommended'],
  overrides: [
    {
      parser: '@typescript-eslint/parser',
      extends: [
        'plugin:@typescript-eslint/recommended',
        'prettier/@typescript-eslint',
        'plugin:import/typescript',
      ],
      plugins: ['@typescript-eslint'],

      files: ['*.ts', '*.tsx'],

      rules: {},
    },
  ],
}
```

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.1.Z` |
| `TypeScript`                | `3.5.3` |
| `ESLint`                    | `6.3.0` |
| `node`                      | `10.16.0` |
| `npm`                       | `6.9.0` |

**Comment by bradzacher** [comment]
I am unable to reproduce this.

I tried against v1.x on astexplorer.net, and against master.

**Comment by Tom910** [comment]
![parsing](https://user-images.githubusercontent.com/2185721/64557720-da98d300-d34a-11e9-8577-e37caa2efc5b.jpg)
I reproduce in astexplorer.net link: https://astexplorer.net/#/gist/2a2e71bec639881726abf4ad539df449/703e114e5e9fda99f48f2af5ca0e39b310984d00

**Comment by bradzacher** [comment]
Looking at the link you sent, that is because it's switching to babel parser, which doesn't understand typescript syntax by default.

If you manage to find a repro case, will gladly reopen this issue.

**Comment by Tom910** [comment]
repository with reproduce - https://github.com/Tom910/eslint-extends-error

The main problem with used extends and overrides. And this case when I create file ts.config.js with ts configuration and `parser `config not change and I get parsing error

**Comment by bradzacher** [comment]
This is not an issue with our parser. I cannot reproduce this with our parser
This is the base babel-eslint parser not working correctly without any config.
ESLint is not applying different parsers to different files, hence babel-eslint is throwing an error.

Why are you mixing and matching parser? You should just be using our parser for everything.

**Comment by Tom910** [comment]
> Why are you mixing and matching parser? You should just be using our parser for everything.

Thanks for your answer. I will test with parsers `@typescript-eslint/parser` for js files. But the main case used a `babel-eslint` it custom syntactic. In my project use mixed files: Babel(js) and TS. Before that, I used `tslint` for ts and `eslint` for js.

**Comment by Tom910** [comment]
I have spent more time to debug eslint combine config and resolved the problem of move js config with separate file

https://github.com/Tom910/eslint-extends-error
